### PR TITLE
Run CI for Java 8

### PR DESCRIPTION
**Pull request by Ndiritu**
related to https://github.com/microsoftgraph/msgraph-sdk-java/issues/1979

### CODE REVIEW

**Comment by Ndiritu** [comment]
@baywet  The SonarQube plugin for Gradle [requires Java 17](https://docs.sonarsource.com/sonarqube/latest/analyzing-source-code/scanner-environment/general-requirements/) to run causing compilation to fail with Java 8.

Should we consider switching to the [GitHub action](https://github.com/marketplace/actions/official-sonarqube-scan) or find a way to tweak the build to not execute sonar tasks then run a separate `./gradlew sonarqube ...` command in the workflows?

**Comment by baywet** [comment]
@Ndiritu that's an avenue we can explore. Another one would be to conditionally include the plugin based on the java version.
Also the android api level is failing for the same reason. Since there's already a plugin in there to validate which APIs we're using, and since we're setting up CI to make sure the other scenarios not being validated by the linter are also being validated, I'm suggesting we revert the changes in that workflow. Thoughts?

**Comment by Ndiritu** [comment]
> @Ndiritu that's an avenue we can explore. Another one would be to conditionally include the plugin based on the java version. Also the android api level is failing for the same reason. Since there's already a plugin in there to validate which APIs we're using, and since we're setting up CI to make sure the other scenarios not being validated by the linter are also being validated, I'm suggesting we revert the changes in that workflow. Thoughts?

@baywet I opted to conditionally include the plugin based on the JVM version since the `spotless` plugin also has a min. requirement of Java 11 to execute.

If I understand correctly, you're suggesting that we add Java 8 CI checks to the linting workflow so that we cover both Android & Java 8 checks there?

Also, Java 8 build checks now fail at the tests. Should we downgrade the junit mockito libs & use Java-8 friendly syntax for tests as well? ie.. failures bc of using `var` keyword

**Review by baywet — CHANGES_REQUESTED** [review]

**Comment by Ndiritu** [comment]
Build workflow now [fails ](https://github.com/microsoft/kiota-java/actions/runs/9581820670/job/26419457608?pr=1259#step:6:256)for Java-8 incompatibilities

File: settings.gradle [hunk 1]
```diff
@@ -1,5 +1,12 @@
+pluginManagement {
+  plugins {
+    if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17)) {
```
[the remark this answers is not in the document]
Comment (Ndiritu) [reply settings.gradle]: Switched to adding sub-project configured for compilation with Java 8

**Review by baywet — APPROVED** [review]

**Pull request merged**